### PR TITLE
Buildkite deploy

### PR DESCRIPTION
**Pull request by jsanz**
Adding more logic to the Buildkite pipeline to include the deployment of the assets to Google Cloud buckets.

* `catalog-info.xml` is updated to include references to components Landing Page depends on, and a setting to include builds when tags are pushed to the repository.
* Current deploy scripts will exit without doing anything.
* The build of the assets has been moved to `.buildkite/scripts/build.sh` for better logging.
* `.buildkite/hooks/pre-command` authenticates with Google Cloud and sets the environment for steps that deploy assets
* `deploy-staging` step in the pipeline will run when the branch of the commit triggering the build is `master` or any of the version branches (`v7.17`, `v8.x`) and there is no tag associated to this executtion. This step runs the command `upload.sh` described below.
* Similarly to the EMS File Service, for production deployments a block step `should-deploy` is set up to make a member of the team to unlock the upload of the new assets, so we can track the process closely. We may remove this block later.
* If `should-deploy` is accepted, there is a git tag, and the branch is whitelisted (`v7.17`, `v8.x`, etc.) the `deploy-production` step will execute:
  * `.buildkite/scripts/upload.sh` sets the destination bucket according to the environment (staging or production), uploads the content of the build step, and if the build branch is equal to the `ROOT_BRANCH` variable it will also synchronize the branch against the root folder at <maps.elastic.co>.
  * `.buildkite/scripts/archive.sh` runs only on production deployments and will archive the current content of the production bucket **before** uploading the new assets.
* Most of the core logic has been copied from the current `.deployStaging`.

### CODE REVIEW

**Comment by jsanz** [comment]
@nickpeihl I've marked this for review but I don't expect this to happen before mid May :slightly_smiling_face:

File: .buildkite/hooks/pre-command [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#
+# Parameters to export:
+# - STAGING_BUCKET: GCP bucket storing the staging copy for testing
+# - PRODUCTION_BUCKET: GCP bucket exposed at maps.elastic.co
+# - ARCHIVE_BUCKET: GCP bucket with the archives of previous builds
+# - ROOT_BRANCH
+
+GCS_VAULT_SECRET_PATH="secret/ci/elastic-ems-landing-page/gce/elastic-bekitzur/service-account/maps-landing"
+PREFIX="elastic-bekitzur-maps-landing-page"
+ROOT_BRANCH='v8.6'
```
Comment (nickpeihl) [inline .buildkite/hooks/pre-command:54]: Should this be `v8.7`?

**Review by nickpeihl — APPROVED** [review]
lgtm! code review

File: .buildkite/hooks/pre-command (hunk 1 above)
Comment (jsanz) [reply .buildkite/hooks/pre-command:54]: Yes, I just did not want to put that change here, but yeah let's also do it here.

**Comment by jsanz** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|v8.9|[](https://github.com/elastic/ems-landing-page/pull/570)|
|✅|v8.8|[](https://github.com/elastic/ems-landing-page/pull/571)|
|✅|v8.7|[](https://github.com/elastic/ems-landing-page/pull/572)|
|✅|v7.17|[](https://github.com/elastic/ems-landing-page/pull/573)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**